packages/gatsby-source-contentful/src/__tests__/gatsby-node.js

```diff
@@ -0,0 +1,346 @@
+// This is more an integration test than it is a unit test. We try to mock as little as we can
+
+jest.mock(`../fetch`)
+jest.mock(`gatsby-core-utils`, () => {
+  return {
+    createContentDigest: () => `contentDigest`,
+  }
+})
+
+const gatsbyNode = require(`../gatsby-node`)
+const fetch = require(`../fetch`)
+const normalize = require(`../normalize`)
+
+const startersBlogFixture = require(`../__fixtures__/starter-blog-data`)
+
+describe(`gatsby-node`, () => {
+  const actions = {}
+  const store = {}
+  const cache = {}
+  const getCache = jest.fn()
+  const reporter = {
+    info: jest.fn(),
+  }
+  const createNodeId = jest.fn(value => value)
+  let currentNodeMap
+  let getNodes = () => Array.from(currentNodeMap.values())
+  let getNode = id => currentNodeMap.get(id)
+
+  beforeEach(() => {
+    fetch.mockClear()
+    currentNodeMap = new Map()
+    actions.createNode = jest.fn(async node => {
+      node.internal.owner = `gatsby-source-contentful`
+      // don't allow mutations (this isn't traversing so only top level is frozen)
+      currentNodeMap.set(node.id, Object.freeze(node))
+    })
+    actions.deleteNode = ({ node }) => {
+      currentNodeMap.delete(node.id)
+    }
+    actions.touchNode = jest.fn()
+    actions.setPluginStatus = jest.fn()
+    store.getState = jest.fn(() => {
+      return {
+        status: {},
+      }
+    })
+  })
+
+  it(`should create nodes from initial payload`, async () => {
+    fetch.mockImplementationOnce(() => startersBlogFixture.initialSync)
+
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
```

Comment:
there is no assertion here - is this by design? so just checking if this won't throw